pkg/cmd/service/service.go

```diff
@@ -20,59 +18,126 @@ import (
 	"github.com/openfga/openfga/storage/caching"
 	"github.com/openfga/openfga/storage/memory"
 	"github.com/openfga/openfga/storage/postgres"
+	"github.com/spf13/viper"
 	"google.golang.org/grpc"
 )
 
 var (
-	ErrInvalidGRPCTLSConfig = errors.New("'OPENFGA_GRPC_TLS_CERT_PATH' and 'OPENFGA_GRPC_TLS_KEY_PATH' env variables must be set")
-	ErrInvalidHTTPTLSConfig = errors.New("'OPENFGA_HTTP_GATEWAY_TLS_CERT_PATH' and 'OPENFGA_HTTP_GATEWAY_TLS_KEY_PATH' env variables must be set")
+	ErrInvalidGRPCTLSConfig = errors.New("'grpc.tls.cert' and 'grpc.tls.key' configs must be set")
+	ErrInvalidHTTPTLSConfig = errors.New("'http.tls.cert' and 'http.tls.key' configs must be set")
 )
 
+type DatabaseConfig struct {
+	Engine string
+	URI    string
+}
+
+type GRPCConfig struct {
+	Enabled bool
+	Port    int
+	TLS     TLSConfig
+}
+
+type HTTPConfig struct {
+	Enabled bool
+	Port    int
+	TLS     TLSConfig
+}
+
+type TLSConfig struct {
+	Enabled  bool
+	CertPath string
+	KeyPath  string
+}
+
+type AuthnConfig struct {
+	Method                   string
+	*AuthnOIDCConfig         `mapstructure:"oidc"`
+	*AuthnPresharedKeyConfig `mapstructure:"preshared"`
+}
+
+type AuthnOIDCConfig struct {
+	Issuer   string
+	Audience string
+}
+
+type AuthnPresharedKeyConfig struct {
+	Keys []string
+}
+
+type LogConfig struct {
+	Format string
+}
+
 type Config struct {
 	// If you change any of these settings, please update the documentation at https://github.com/openfga/openfga.dev/blob/main/docs/content/intro/setup-openfga.mdx
-	DatastoreEngine               string `default:"memory" split_words:"true"`
-	DatastoreConnectionURI        string `split_words:"true"`
-	DatastoreMaxCacheSize         int    `default:"100000" split_words:"true"`
-	ServiceName                   string `default:"openfga" split_words:"true"`
-	HTTPPort                      int    `default:"8080" split_words:"true"`
-	RPCPort                       int    `default:"8081" split_words:"true"`
-	MaxTuplesPerWrite             int    `default:"100" split_words:"true"`
-	MaxTypesPerAuthorizationModel int    `default:"100" split_words:"true"`
+
+	DatabaseConfig `mapstructure:"database"`
```

Comment:
Since these are in the `Config` struct does it make sense to drop the `Config` postfix. Then we can drop the `mapstructure` tag as well.

Reply:
Oh, I see, you want to embed the configs in this struct. In general I like being more explicit in which case we could have
```
Database DatabaseConfig
```
here, then the configuration stuctures would have clear names. 

In this case, how do you avoid collisions with `Port` and `Enabled` for example?

Reply:
The collisions are avoided because the embedded struct is scoped by it's defining type.

For example,

```
type DatabaseConfig struct {
    Port int
}

type OtherConfig struct {
    Port int
}

type Config struct {
    DatabaseConfig
    OtherConfig
}
```

the `Port` field is actually namespaced as `Config.DatabaseConfig.Port` and `Config.OtherConfig.Port`, so there are no collisions.

Reply:
vote: option 1

Reply:
👍 sounds good. Thanks for the vote!